File: java-maven/hello-binary/src/main/java/com/example/Hello.java
package com.example;

/**
 * Minimal Java executable for the M40 ``java-maven/hello-binary``
 * fixture. The pom.xml's ``maven-jar-plugin`` configuration sets the
 * jar's ``Main-Class`` manifest attribute to ``com.example.Hello`` so
 * the produced ``target/hello-1.0.jar`` runs via
 * ``java -jar target/hello-1.0.jar``.
 *
 * Imports nothing beyond ``java.lang`` (stdlib) so the build needs no
 * external Maven dependencies — keeping the M40 offline-mode contract
 * (``mvn package -o``) trivially satisfiable without a warm-step.
 */
public class Hello {
    public static void main(String[] args) {
        System.out.println("hello from java-maven-hello-binary");
    }
}

File: kotlin-gradle/hello-binary/src/main/kotlin/Hello.kt
/**
 * Minimal Kotlin executable for the M41 ``kotlin-gradle/hello-binary``
 * fixture. The build.gradle.kts's ``application.mainClass`` is set to
 * ``HelloKt`` — Kotlin compiles a top-level ``fun main()`` into a
 * synthetic class named ``<Filename>Kt`` (here ``HelloKt``), and the
 * ``jar`` task's manifest pins ``Main-Class: HelloKt`` so the produced
 * ``build/libs/hello-1.0.jar`` runs via
 * ``java -jar build/libs/hello-1.0.jar``.
 *
 * Imports nothing beyond the Kotlin stdlib (which Gradle pulls from
 * ``~/.gradle/caches/`` during the offline build) so the build needs
 * no external runtime dependencies — keeping the M41 offline-mode
 * contract (``gradle build --offline``) trivially satisfiable after a
 * one-time warm-step.
 */
fun main() {
    println("hello from kotlin-gradle-hello-binary")
}
